src/providers/fusionauth.ts

```diff
@@ -0,0 +1,52 @@
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
+
+// This is the default openid signature returned from FusionAuth
+// it can be customized using lambda functions https://fusionauth.io/docs/v1/tech/lambdas/
+export interface FusionAuthProfile {
+  aud: string
+  exp: number
+  iat: number
+  iss: string
+  sub: string
+  jti: string
+  authenticationType: string
+  email: string
+  email_verified: boolean
+  preferred_username: string
+  at_hash: string
+  c_hash: string
+  scope: string
+  sid: string
+}
+
+export default function FusionAuth<
+  P extends Record<string, any> = FusionAuthProfile
+>(
+  // tenantId only needed if there is more than one tenant configured on the server
+  options: OAuthUserConfig<P> & { tenantId?: string }
+): OAuthConfig<P> {
+  return {
+    id: "fusionauth",
+    name: "FusionAuth",
+    type: "oauth",
+    wellKnown: !options?.tenantId
+      ? `${options.issuer}/.well-known/openid-configuration`
+      : `${options.issuer}/.well-known/openid-configuration?tenantId=${options.tenantId}`,
+    jwks_endpoint: `${options.issuer}/.well-known/jwks.json`,
```

Comment:
I wonder if this might be unnecessary/extracted from the `openid-configuration` endpoint? :thinking:

Reply:
I think so yes, the response payload from the [docs](https://fusionauth.io/docs/v1/tech/oauth/endpoints/#openid-configuration) returns it, but don't know how could we use that?

Reply:
just remove it, I believe `openid-client` will take care of the rest.😉 do test locally though, please.

Reply:
So it works, but for some reason the `iframe` in `api-example` is returning `null`; but the call to `/api/session` works correctly (see image
![Screenshot_20211207_113618](https://user-images.githubusercontent.com/6154350/145022808-d74b2792-6efb-41b6-9c56-1efc567c4637.png)
).

Reply:
Ah never mind, I'm using a self-signed certificate to fake SSL so that was interfering with it; setting `NODE_TLS_REJECT_UNAUTHORIZED` to `0` makes it work. You think this is good to go now? @balazsorban44 
![Screenshot 2021-12-08 at 12-55-23 https localhost](https://user-images.githubusercontent.com/6154350/145211858-87ac7bde-1f08-412a-ba24-82c6fc144135.png)
.